c/misra/src/rules/RULE-5-2/IdentifiersDeclaredInTheSameScopeNotDistinct.ql

```diff
@@ -0,0 +1,38 @@
+/**
+ * @id c/misra/identifiers-declared-in-the-same-scope-not-distinct
+ * @name RULE-5-2: Identifiers declared in the same scope and name space shall be distinct
+ * @description Using nondistinct identifiers results in undefined behaviour.
+ * @kind problem
+ * @precision very-high
+ * @problem.severity warning
+ * @tags external/misra/id/rule-5-2
+ *       correctness
+ *       maintainability
+ *       readability
+ *       external/misra/obligation/required
+ */
+
+import cpp
+import codingstandards.c.misra
+import codingstandards.cpp.Identifiers
+
+from InterestingIdentifiers d, InterestingIdentifiers d2
+where
+  not isExcluded(d, Declarations5Package::identifiersDeclaredInTheSameScopeNotDistinctQuery()) and
+  not isExcluded(d2, Declarations5Package::identifiersDeclaredInTheSameScopeNotDistinctQuery()) and
+  //this rule does not apply if both are external identifiers
+  //that is covered by RULE-5-3
+  not (
```

Comment:
I think that the test-case for Rule 5.2 should include a test case for one of the identifiers being a macro identifier in order to address the exception for macro identifiers. It seems okay if Rule 5.2 does not explicitly include that logic other than implicitly excluding macro identifiers from `InterestingIdentifiers`, but that differentiation should be clearly documented and tested.

Reply:
makes sense to me, I also added an additional testcase, a nondistinct but external identifier that also would be compliant via exception, thanks for the testcase improvement!